src/builder/QuarkBuilder.sol

```diff
@@ -74,22 +73,28 @@ contract QuarkBuilder {
 
         // XXX confirm that the user is not withdrawing beyond their limits
 
-        assertFundsAvailable(
-            repayIntent.chainId, repayIntent.assetSymbol, repayIntent.amount, chainAccountsList, payment
-        );
+        bool isMaxRepay = repayIntent.amount == type(uint256).max;
+        bool useQuotecall = isMaxRepay;
+
+        uint256 repayAmount;
+        if (isMaxRepay) {
+            repayAmount += Accounts.totalBorrowForAccount(
+                chainAccountsList, repayIntent.chainId, repayIntent.comet, repayIntent.repayer
+            );
+            // XXX apply buffer to repayAmount
```

Comment:
Not sure what our buffer logic should be.

Maybe we add either 1% of the `repayAmount` or some `MIN_BUFFER` value, whichever is larger?

Reply:
That sounds good to me 👍 
Or we can just do % increase as well,  since interest accrue is purely based on %.

Reply:
Yeah % makes sense. Something small like 0.1% is probably already plenty for most situations, but this really depends on the current borrow rate, which might be too complex to build in for now. Maybe leave that as a TODO.

Just to highlight how large 0.1% already is, let's use 20% borrow APR as an example. If we convert this to a daily rate, we get 20/365=0.0548%. So 0.1% pays for 2 days of a borrow that has 20% borrow APR. That seems plenty if we expect to repay to happen in an hour or so.